Add suse support

andy-dufour opened the pull request:
Currently when you try to run an omnibus installer on Suse or SLES you receive a NilClass error on runit services. This defines the correct runit supervisor, at least on SLES11. We will probably need to split this eventually based on Suse version as I believe SLES12 is not SysV.

andy-dufour commented:
@chef-cookbooks/engineering-services for review

irvingpop commented:
:+1: makes total sense - @andy-dufour could you please provide evidence of testing?

andy-dufour commented:
The Fedora v20 fauxhai data was changed which fubar'd the fedora tests. I've fixed them to pass correctly on Fedora v20.

Rubocop was complaining but one of the replacement rules it suggested didn't exist yet - had to bump to 0.37. 

Foodcritic was complaining about some conditionals in the runit definition, I added an exemption for that one.

And, I've split out SuSE 11 and 12 logic so we get the right provider on each system, and added tests.

Passes the spec tests and I've done some manual testing.

irvingpop commented:
awesome! :+1:

tas50 commented:
@andy-dufour Out of curiosity, what in the Fedora Fauxhai data broke things?  I was the one that regenerated all the Fauxhai data.  It was missing tons of data before.

andy-dufour commented:
@tas50 Yup, you got it right - it was pretty far off before from what I could see and you fixed it.

The new fauxhai data has "init_package": "systemd" in the data:

https://github.com/customink/fauxhai/blob/master/lib/fauxhai/platforms/fedora/20.json#L306

Our spec tests expected Fedora to hit systemv, and instead we started hitting systemd once that ohai data was present:

https://github.com/chef-cookbooks/enterprise-chef-common/blob/master/libraries/provider_component_runit_supervisor_systemd.rb#L7-L9

sdelano commented on recipes/runit.rb:
```diff
@@ -27,7 +27,7 @@
 component_runit_supervisor node['enterprise']['name'] do
   ctl_name node[node['enterprise']['name']]['ctl_name'] ||
-    "#{node['enterprise']['name']}-ctl"
+           "#{node['enterprise']['name']}-ctl"
```
whats with all the extra whitespace? Is this a `rubocop` thing?

andy-dufour replied:
Yeah. Rubocop wants all attributes on new lines aligned or some such thing..

sdelano commented:
LGTM :+1: - just one whitespace nitpick but I'm in a good mood today, so ¯\_(ツ)_/¯

andy-dufour commented:
Unfortunately, I don't have the big green merge button on this project. If someone could hit it that would be awesome!
